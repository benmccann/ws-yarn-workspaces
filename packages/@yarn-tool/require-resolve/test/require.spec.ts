import requireResolveExtra, { isErrorModuleNotFound } from '../index';

describe(`describe`, () =>
{
	const tsdx_path = requireResolveExtra('tsdx').result;

	console.dir(tsdx_path)

	test(`ts-jest`, () =>
	{

		let actual = requireResolveExtra('ts-jest', {
			includeGlobal: true,
			includeCurrentDirectory: true,
			paths: [
				tsdx_path,
			],
		});

		console.dir(actual);

		expect(actual).toMatchSnapshot({
			result: expect.any(String),
			error: void 0
		});

	});

	test(`typescript`, () =>
	{

		let actual = requireResolveExtra('typescript', {
			includeGlobal: true,
			includeCurrentDirectory: true,
			paths: [
				tsdx_path,
			],
		});

		console.dir(actual);

		expect(actual).toMatchSnapshot({
			result: expect.any(String),
			error: void 0
		});

	});

	test(`MODULE_NOT_FOUND`, () =>
	{

		let actual = requireResolveExtra('@typescript', {
			includeGlobal: true,
			includeCurrentDirectory: true,
			paths: [
				tsdx_path,
			],
		});

		console.dir(actual);

		expect(actual).toHaveProperty('error.code', 'MODULE_NOT_FOUND');

		expect(isErrorModuleNotFound(actual.error)).toBeTruthy();

		expect(actual.result).toBeUndefined();

	});

})
