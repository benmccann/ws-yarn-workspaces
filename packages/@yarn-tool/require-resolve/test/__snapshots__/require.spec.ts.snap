// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`describe ts-jest 1`] = `
{
  "error": undefined,
  "result": Any<String>,
}
`;

exports[`describe typescript 1`] = `
{
  "error": undefined,
  "result": Any<String>,
}
`;
